src/Razor/test/Microsoft.VisualStudio.Razor.Integration.Test/InProcess/WorkspaceInProcess.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.OperationProgress;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.Razor.Integration.Test.InProcess
+{
+    internal class WorkspaceInProcess : InProcComponent
+    {
+        public WorkspaceInProcess(TestServices testServices)
+            : base(testServices)
+        {
+        }
+
+        public Task WaitForAsyncOperationsAsync(string featuresToWaitFor, CancellationToken cancellationToken)
+            => WaitForAsyncOperationsAsync(featuresToWaitFor, waitForWorkspaceFirst: true, cancellationToken);
+
+        public async Task WaitForAsyncOperationsAsync(string featuresToWaitFor, bool waitForWorkspaceFirst, CancellationToken cancellationToken)
+        {
+            if (waitForWorkspaceFirst || featuresToWaitFor == FeatureAttribute.Workspace)
+            {
+                await WaitForProjectSystemAsync(cancellationToken);
+            }
+
+            // TODO: This currently no-ops on the FeaturesToWaitFor portion
```

Comment:
As we expand our tests and add new things that we need to await the lack of this may make it hard to create reliable tests. It'll probably be the first expansion we'll need, but I've left it out since the PR is already quite large and I was able to target a couple tests which didn't need it.